dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaCMFHostsTask.java

```diff
@@ -72,11 +81,21 @@ protected Void doRun(
         String hostPerClusterUrl =
             baseURI + "/cmf/hardware/hosts/hostsOverview.json?clusterId=" + cluster.getId();
 
-        try (CloseableHttpResponse hosts = httpClient.execute(new HttpGet(hostPerClusterUrl))) {
-          JsonNode jsonNode = objectMapper.readTree(hosts.getEntity().getContent());
-          writer.write(jsonNode.toString());
+        try (CloseableHttpResponse hostsResponse =
+            httpClient.execute(new HttpGet(hostPerClusterUrl))) {
+          JsonNode hostsJson = objectMapper.readTree(hostsResponse.getEntity().getContent());
+          String stringifiedHosts = hostsJson.toString();
+          apiHosts = objectMapper.readValue(stringifiedHosts, ApiHostListDto.class);
+          writer.write(stringifiedHosts);
           writer.write('\n');
         }
+
+        for (ApiHostDto apiHost : apiHosts.getItems()) {
+          hosts.add(ClouderaHostDTO.create(apiHost.getId(), apiHost.getName()));
+        }
+      }
+      if (!handle.isHostInitialized()) {
```

Comment:
This block should be atomic and thread safe.

Reply:
Updated